fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2384,7 +2384,7 @@ public ChangedTransactionDetail disburse(final AppUser currentUser, final JsonCo
          **/
 
         if (isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()
-                        && (isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1 || !isMultiDisburmentLoan())) {
+                        && ((isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1) || !isMultiDisburmentLoan())) {
```

Comment:
Wouldn't this be better & clearer like this, instead:

    if ((isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()
        && isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1) || !isMultiDisburmentLoan()) {

Reply:
OK will try this but the parentheses grouping might change the logic of the check